homeassistant/components/apcupsd/config_flow.py

```diff
@@ -52,9 +52,8 @@ async def async_step_user(
 
         # Test the connection to the host and get the current status for serial number.
         coordinator = APCUPSdCoordinator(self.hass, host, port)
-
         await coordinator.async_request_refresh()
-        await self.hass.async_block_till_done()
```

Comment:
This will wait for everything, not just the integration so it will likely never finish once startup has completed.

Reply:
Huh, that makes sense now! Thanks for the explanations!